File: demo/src/main/java/com/nshmura/strictmodenotifier/demo/LeakedBroadcastReceiver.java
package com.nshmura.strictmodenotifier.demo;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Handler;

public class LeakedBroadcastReceiver extends BroadcastReceiver {

  public LeakedBroadcastReceiver() {
  }

  @Override public void onReceive(Context context, Intent intent) {
    // Post a message and delay its execution for 10 minutes.
    new Handler().postDelayed(new Runnable() {
      @Override public void run() {
      }
    }, 1000 * 60 * 10);
  }
}

File: testapp/src/main/java/io/github/juanimoli/strictmode/notifier/receiver/LeakedBroadcastReceiver.kt
package io.github.juanimoli.strictmode.notifier.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Handler
import android.os.Looper

class LeakedBroadcastReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        // Post a message and delay its execution for 10 minutes.
        Handler(Looper.getMainLooper()).postDelayed({}, (1000 * 60 * 10).toLong())
    }
}
